flutter_keyboard_visibility/lib/src/ui/keyboard_dismiss_on_tap.dart

```diff
@@ -7,27 +7,69 @@ import 'package:flutter/widgets.dart';
 /// tree and when the user taps outside of a focused widget,
 /// the focus will be removed and the keyboard will be hidden.
 class KeyboardDismissOnTap extends StatelessWidget {
+  /// Used to control if clicking a widget inheriting GestureDetector will
+  /// dismiss the keyboard or not
+  ///
+  /// By default GestureDetectors are ignored
+  final bool ignoreGestureDetectors;
+
+  /// This is used to control if the next dismiss is allowed
+  static bool _dismissible = true;
+
   const KeyboardDismissOnTap({
     Key? key,
     required this.child,
+    this.ignoreGestureDetectors = true,
   }) : super(key: key);
 
   final Widget child;
 
   void _hideKeyboard(BuildContext context) {
-    final currentFocus = FocusScope.of(context);
+    Future.delayed(Duration.zero, () {
```

Comment:
I don't think using Future.delayed is a great solution, hopefully we can avoid with the InheritedWidget changes.